From 626359b8abd1b8c08fb1896ba60abda0b694f252 Mon Sep 17 00:00:00 2001
From: Vlad Yasevich <vyasevic@redhat.com>
Date: Wed, 10 Sep 2014 15:16:15 -0400
Subject: [net] bridge: Use RCU_INIT_POINTER(x, NULL) in br_vlan.c

Message-id: <1410362181-9327-3-git-send-email-vyasevic@redhat.com>
Patchwork-id: 93478
O-Subject: [RHEL7 PATCH 2/8] net/bridge: Use RCU_INIT_POINTER(x, NULL) in br_vlan.c
Bugzilla: 1139809
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>
RH-Acked-by: Flavio Leitner <fbl@redhat.com>

Bugzilla: 1139809
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7936510

commit cd18721e527270f9afa5c8daabfb6403b01e3a92
From: Monam Agarwal <monamagarwal123@gmail.com>
Date: Mon, 24 Mar 2014 00:41:13 +0530

    net/bridge: Use RCU_INIT_POINTER(x, NULL) in br_vlan.c

    This patch replaces rcu_assign_pointer(x, NULL) with RCU_INIT_POINTER(x, NULL)

    The rcu_assign_pointer() ensures that the initialization of a structure
    is carried out before storing a pointer to that structure.
    And in the case of the NULL pointer, there is no structure to initialize.
    So, rcu_assign_pointer(p, NULL) can be safely converted to RCU_INIT_POINTER(p, NULL)

    Signed-off-by: Monam Agarwal <monamagarwal123@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Vladislav Yasevich <vyasevic@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/bridge/br_vlan.c b/net/bridge/br_vlan.c
index 210b348..c808012 100644
--- a/net/bridge/br_vlan.c
+++ b/net/bridge/br_vlan.c
@@ -99,9 +99,9 @@ static int __vlan_del(struct net_port_vlans *v, u16 vid)
  v->num_vlans--;
  if (bitmap_empty(v->vlan_bitmap, VLAN_N_VID)) {
   if (v->port_idx)
-   rcu_assign_pointer(v->parent.port->vlan_info, NULL);
+   RCU_INIT_POINTER(v->parent.port->vlan_info, NULL);
   else
-   rcu_assign_pointer(v->parent.br->vlan_info, NULL);
+   RCU_INIT_POINTER(v->parent.br->vlan_info, NULL);
   kfree_rcu(v, rcu);
  }
  return 0;
@@ -113,9 +113,9 @@ static void __vlan_flush(struct net_port_vlans *v)
  v->pvid = 0;
  bitmap_zero(v->vlan_bitmap, VLAN_N_VID);
  if (v->port_idx)
-  rcu_assign_pointer(v->parent.port->vlan_info, NULL);
+  RCU_INIT_POINTER(v->parent.port->vlan_info, NULL);
  else
-  rcu_assign_pointer(v->parent.br->vlan_info, NULL);
+  RCU_INIT_POINTER(v->parent.br->vlan_info, NULL);
  kfree_rcu(v, rcu);
 }
 
-- 
1.7.1